### PR TITLE
ThemesSiteSelectorModal: Refactor

### LINKED ISSUE #1620: Themes: Refactor theme-options
In order to get rid of some more of the ugliness that is `my-sites/themes/main.jsx`, we'll need to refactor `theme-options`. In `main.jsx`, we're now passing props `selectedAction` (a string), `actions`, and `getOptions` to `ThemesSiteSelectorModal`. Ideally, I think it should be just a `selectedAction` function -- something that is currently computed inside `ThemesSiteSelectorModal`. Urgh.

(From https://github.com/Automattic/wp-calypso/pull/840#issuecomment-164563182)

### PR DESCRIPTION
**Pull request by ockham**
`ThemeSiteSelectorModal` is used when an action (typically from the 'More' button) is called in multi-site mode, and the user has to select on which site to apply that action. Turns out that there are only three such actions these days: activate, customize, and purchase. With this knowledge, we can just pass those actions and their labels and headers to `ThemeSiteSelectorModal` instead of `getButtonOptions`, which returns way more stuff which is only needed for the 'More' button.

This PR thus factors that basic information into a separate module (`action-labels`), and returns that information as well as the return value from `getButtonOptions` into a keyed object so we can leverage key lookup (instead of `find`ing stuff).

No visual changes. To test -- check that the theme showcase works as before (especially multi-site). Also check the logged-out signup flow since it also uses `ThemesList`.

Closes #1620

### CODE REVIEW

File: client/my-sites/themes/themes-site-selector-modal.jsx [hunk 1]
```diff
@@ -13,61 +11,73 @@ import partial from 'lodash/function/partial';
...
 		} );
 	},
 
-	getUrl( action, theme, site, getOptions ) {
-		var option = find( getOptions( site, theme ), { name: action } );
-
-		if ( option ) {
-			return option.url;
-		}
+	getActionText( name ) {
```
Comment (seear) [inline client/my-sites/themes/themes-site-selector-modal.jsx]: To de-dupe this for now we could move it back to the `theme-options` module as a separate function (maybe adding the rest of the strings) and have `getButtonOptions()` call it as well.

Comment (ockham) [inline client/my-sites/themes/themes-site-selector-modal.jsx]: Yeah, sounds good. I'll have a stab at that...

File: client/my-sites/themes/theme-options.js [hunk 2]
```diff
@@ -3,21 +3,77 @@
...
+			mapValues(
+				pick(
+					buttonOptions,
+					option => ! ( option.isHidden && option.isHidden( site, theme, isLoggedOut ) )
+				), appendLabelAndHeader
+			), appendUrl
+		), appendAction
+	);
+
+	function appendLabelAndHeader( option, name ) {
```
Comment (seear) [inline client/my-sites/themes/theme-options.js]: nice solution ;)

File: client/my-sites/themes/theme-options.js [hunk 3]
```diff
@@ -3,21 +3,77 @@
...
+		hasUrl: true
+	},
+	support: {
+		hasUrl: true,
+		isHidden: site => site && site.jetpack // We don't know where support docs for a given theme on a self-hosted WP install are.
+	},
+};
+
+export function getButtonOptions( site, theme, isLoggedOut, actions, setSelectedTheme, togglePreview ) {
+	return mapValues(
```
Comment (seear) [inline client/my-sites/themes/theme-options.js]: Should we go with something a little less nested? e.g:
```
	let options = pick( buttonOptions, option => ! ( option.isHidden && option.isHidden( site, theme, isLoggedOut ) ) );
	options = mapValues( options, appendLabelAndHeader );
	options = mapValues( options, appendUrl );
	options = mapValues( options, appendAction );
	return options;
```

**Comment by seear** [comment]
Tested: single site more button options; multisite more button options, modals; /start; analytics. All seems fine. Lets get this merged soon :+1:

File: client/my-sites/themes/theme-options.js (hunk 3 above)
Comment (ockham) [inline client/my-sites/themes/theme-options.js]: Good call. 525cc2b4ab1d046d6f742b687c4d8cfaa18e0cc3

**Pull request merged**